### PR TITLE
Passes all test cases in JupyterNotebook

### PR DESCRIPTION
**Pull request by FirdausChoudhury**

### CODE REVIEW

**Comment by BirkhoffG** [comment]
Can you run `nbdev_export` and `nbdev_clean`, and commit again?

**Comment by FirdausChoudhury** [comment]
<p>Ran them and commited</p>

File: nbs/00_data_preprocessor.ipynb [hunk 1]
```diff
@@ -54,25 +54,24 @@
    "metadata": {},
```
Comment (BirkhoffG) [inline nbs/00_data_preprocessor.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #9.            if self.min_ == self.max_:
</pre><p>In general, to prevent the denominator is close to 0, what we can do is to add a small number in the denominator (e.g., 1e-5).</p><p> </p><p>Writing <code>if</code> might not prevent corner cases. If  self.min_ = self.max_ + 1e-6, sometimes it might also lead to error, because the denominator is close to 0, and get round to 0.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/jax-practice/pull/1/discussion/#comment-1367572542'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='6'/>

Comment (BirkhoffG) [inline nbs/00_data_preprocessor.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #13.            encoder = skp.OneHotEncoder(sparse_output=False)
</pre><p>This is not wrong, but could you write the <code>OneHotEncoder</code>  without using sklearn? Please rewrite the transform/inverse_transform functions here.</p><p> </p><p>Hint: check <code>jax.nn.one_hot</code>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/jax-practice/pull/1/discussion/#comment-1367572543'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (BirkhoffG) [inline nbs/00_data_preprocessor.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #25.                    xs = xs.astype(np.float64)
</pre><p>I would simply write <code>xs = xs.astype(xs.dtype)</code>. No need for an <code>if</code>   satetment.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/jax-practice/pull/1/discussion/#comment-1367572545'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (BirkhoffG) [inline nbs/00_data_preprocessor.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #9.            for feature in self.features:
</pre><p>This is correct, but you can write it more concise by using <code>enumerate</code> </p><p> </p><p><code></code><code>python</p><p>for i, feature in enumerate(self.features):</p><p>    ...</p><p></code><code></code> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/BirkhoffG/jax-practice/pull/1/discussion/#comment-1367572546'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='11'/>

**Comment by BirkhoffG** [comment]
@FirdausChoudhury Thanks for your PR. Could you make changes based on my comments above?

In addition, you have deleted many of the `#| export` directives. This directive label the block to be exported to `jax_practice/data_preprocessor.py`, so that we can use code written in the jupyter notebook.

After making these changes, please run `nbdev_export` and `nbdev_clean` before commit.

**Review by BirkhoffG — CHANGES_REQUESTED** [review]
Thanks for your PR. Could you make changes based on my comments above?

In addition, you have deleted many of the `#| export` directives. This directive label the block to be exported to `jax_practice/data_preprocessor.py`, so that we can use code written in the jupyter notebook.

After making these changes, please run `nbdev_export` and `nbdev_clean` before commit.

**Comment by FirdausChoudhury** [comment]
<p>Thanks for the feedback. I'll get them implemented and commit them</p>

File: nbs/00_data_preprocessor.ipynb (hunk 1 above)
Comment (FirdausChoudhury) [reply nbs/00_data_preprocessor.ipynb]: <p>change completed</p>

Comment (FirdausChoudhury) [reply nbs/00_data_preprocessor.ipynb]: <p>resolved</p>

Comment (FirdausChoudhury) [reply nbs/00_data_preprocessor.ipynb]: <p>Completed</p>

**Comment by BirkhoffG** [comment]
Good work @FirdausChoudhury. I think no further comments.

You have successfully completed this task. Congratulations!
